run-cafmaker/gen_cafmaker_cfg.py

```diff
@@ -63,14 +72,27 @@ def main():
         caf_path = args.caf_path
         outf.write(f'nd_cafmaker.CAFMakerSettings.OutputFile: "{caf_path}"\n')
 
-        mlreco_path = get_path(args.base_dir, 'run-mlreco', args.mlreco_name,
-                               'MLRECO_ANA', 'hdf5', args.file_id)
-        outf.write(f'nd_cafmaker.CAFMakerSettings.NDLArRecoFile: "{mlreco_path}"\n')
+        spine_path = get_path(args.base_dir, 'run-mlreco', args.spine_name,
+                               'MLRECO_SPINE', 'hdf5', args.file_id)
+        outf.write(f'nd_cafmaker.CAFMakerSettings.NDLArRecoFile: "{spine_path}"\n')
+
+        if args.minerva_name:
+            minerva_path = get_path(args.base_dir, 'run-minerva', args.minerva_name,
+                                    'DST', 'root', args.file_id)
+            outf.write(f'nd_cafmaker.CAFMakerSettings.MINERVARecoFile: "{minerva_path}"\n')
+
+        if args.tmsreco_name:
+            tmsreco_path = get_path(args.base_dir, 'run-tms-reco', args.tmsreco_name,
+                                    'TMSRECO', 'root', args.file_id)
+            outf.write(f'nd_cafmaker.CAFMakerSettings.TMSRecoFile: "{tmsreco_path}"\n')
 
-        minerva_path = get_path(args.base_dir, 'run-minerva', args.minerva_name,
-                                'DST', 'root', args.file_id)
-        outf.write(f'nd_cafmaker.CAFMakerSettings.MINERVARecoFile: "{minerva_path}"\n')
+        edepsim_path = get_path(args.base_dir, 'run-spill-build', args.edepsim_name,
```

Comment:
Oh, I realize why you moved to "required = true". I think a if here would make sense with a false to the requirement

Reply:
😅 right I wasn't sure whether your intension was to make it a compulsory or non-compulsory argument. Which one would you prefer?

Reply:
2x2 official prod is run with it, but it can run without it so I wouldn"t say it's mandatory, and from what I understand NDLar would run without it for now right?

Reply:
Good call. Fixed in 34c3aba685825b84d8c963e50090ba1d34202089.